src/main/java/me/doubledutch/pikadb/Page.java

```diff
@@ -181,4 +186,81 @@ public void appendData(byte[] data){
 		dirty=true;
 		// System.out.println("currentFill: "+id+" "+currentFill+" "+data.length);
 	}
+}
+
+/**
+ * Adaptive Replacement Cache
+ * Adapted from http://code.activestate.com/recipes/576532/
+ */
+class ARCPageCache implements PageCache{
```

Comment:
Yeah, in Java it's generally preferred to keep one class pr file... the exception are inner classes such as in Variant (though that one is starting to feel messy).